truss/templates/server/model_wrapper.py

```diff
@@ -179,62 +177,19 @@ def _predict_and_post(
         payload: Any,
         headers: Optional[Dict[str, str]] = None,
     ) -> Any:
-        self._predict_lock.acquire()
-        defer_lock_release = False
-        try:
-            response = self.predict(payload, headers)
-            response = self.postprocess(response, headers)
-            if not isinstance(response, Generator):
-                return response
-
-            # Generator response
-            if headers and headers.get("accept") == "application/json":
-                return _convert_streamed_response_to_string(response)
-
-            # Reaching here means streaming response, and need to defer releasing lock
-            defer_lock_release = True
-        finally:
-            if not defer_lock_release:
-                self._predict_lock.release()
-
-        # Streaming response
-        response_queue: Queue = Queue()
-
-        def queue_response_chunks():
```

Comment:
the only reason we needed to do this queuing stuff is to prevent a partial read of the stream from hogging the predict lock. This is no longer an issue.

note that we have an integration test to catch this case still